docker/configure_workers_and_start.py

```diff
@@ -404,11 +405,11 @@ def generate_worker_files(environ, config_path: str, data_dir: str):
         # e.g. federation_reader1
         worker_name = worker_type + str(new_worker_count)
         worker_config.update(
-            {"name": worker_name, "port": worker_port, "config_path": config_path}
+            {"name": worker_name, "port": str(worker_port), "config_path": config_path}
         )
 
         # Update the shared config with any worker-type specific options
-        shared_config.update(worker_config["shared_extra_conf"])
+        shared_config.update(cast(dict, worker_config["shared_extra_config"]))
```

Comment:
mypy has incorrectly determined `WORKERS_CONFIG` to be a `Dict[str, Dict[str, Collection[str]]`.
Instead of `str()` and `cast()`ing, we can type `WORKERS_CONFIG` as a `Dict[str, Dict[str, Any]]` instead.